ls1intum/ArTEMiS
# Bugfix/Correct result shown in result component in exercise detail row

### Checklist

### Motivation and Context



Bug report from https://github.com/ls1intum/ArTEMiS/issues/556.
When going back and forth between the code-editor/exercise detail component and the exercise row component, not the most recent graded result was shown.

I saw a couple of issues in the code that I addressed (see below).

### Description


Issues addressed in this PR:
- The updating result component did not select the most recent graded result for programming-exercises, but the first in the array (which is normally the oldest). Regardless of the exercise type, we should in my opinion always check the result array and get the latest result from it. Otherwise we might forget to add the check for another exercise type in future.
- The websocket subscription was not created if the due date has passed (we should still create it, but filter out ungraded results if necessary as it is an option for the result component to show ungraded results)

While I was at it, I also made some small changes to show ungraded results in the code-editor (see screenshot below).



### Steps for Testing


1. Check exercise row result for a programming-exercise
2. Go to exercise detail component and compare result
3. Go back to exercise row and compare result
-> All results should be the same

### Screenshots



![image](https://user-images.githubusercontent.com/28230611/59469673-cfaad100-8e35-11e9-8818-6b5d518bd57e.png)
